criteria/geode-common/test/org/immutables/criteria/geode/GeodeQueryVisitorTest.java

```diff
@@ -0,0 +1,87 @@
+package org.immutables.criteria.geode;
+
+import static org.immutables.check.Checkers.check;
+import static org.immutables.criteria.matcher.Matchers.toExpression;
+import static org.immutables.criteria.personmodel.PersonCriteria.person;
+
+import org.immutables.criteria.backend.PathNaming;
+import org.immutables.criteria.personmodel.PersonCriteria;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+class GeodeQueryVisitorTest {
+
+    @Test
+    void filter() {
+        check(toOql(person.age.is(18))).is("age = $1");
+        check(toOql(person.age.isNot(18))).is("age != $1");
+
+        check(toOql(person.age.atMost(18))).is("age <= $1");
+        check(toOql(person.age.lessThan(18))).is("age < $1");
+
+        check(toOql(person.age.atLeast(18))).is("age >= $1");
+        check(toOql(person.age.greaterThan(18))).is("age > $1");
+
+        check(toOql(person.age.between(18, 21))).is("(age >= $1) AND (age <= $2)");
+
+        check(toOql(person.address.isPresent())).is("address != null");
+        check(toOql(person.address.isAbsent())).is("address = null");
+
+        check(toOql(person.fullName.isEmpty())).is("fullName = $1");
+        check(toOql(person.fullName.notEmpty())).is("fullName != $1");
+    }
+
+    @Test
+    @Disabled("Disabled until collections supported")
```

Comment:
I'll add support for Iterable queries, String queries, and Aggregations in a follow up